core/src/main/scala/tapir/generic/ValidateMagnoliaDerivation.scala

```diff
@@ -0,0 +1,80 @@
+package tapir.generic
+
+import magnolia.{CaseClass, Magnolia, SealedTrait}
+
+trait ValidateMagnoliaDerivation extends LowPriorityValidators {
+  type Typeclass[T] = Validator[T]
+
+  def combine[T](ctx: CaseClass[Validator, T]): Validator[T] = {
+    ProductValidator(ctx.parameters.map { p =>
+      p.label -> new FieldValidator[T] {
+        override type fType = p.PType
+        override def get(t: T): fType = { p.dereference(t) }
+        override def validator: Typeclass[fType] = p.typeclass
+      }
+    }.toMap)
+  }
+
+  def dispatch[T](ctx: SealedTrait[Validator, T]): Validator[T] = Validator.rejecting
+
+  implicit def gen[T]: Validator[T] = macro Magnolia.gen[T]
+}
+
+trait LowPriorityValidators {
+  def fallback[T]: Validator[T] = Validator.passing
+}
+
+trait Validator[T] { outer =>
+  def validate(t: T): Boolean
```

Comment:
Would it be more useful for this to return a cats `Validated` or `ValidatedNel`? That way the `Validate` decoder can expose specific errors.

Reply:
yes we should probably return some more information. `Validated` would be good though it would mean adding cats as a dependency. So I'm not sure about that yet :)

Reply:
Could we just copy `Validated` from cats?

Reply:
maybe let's start with a simple ValidationError(msg: String)?